arch/armv7m/jmp.S

```diff
@@ -62,24 +62,29 @@ setjmp:
 	mov r3, r0
 	pop {r0, lr}
 	/* set jump address and stack pointer */
-	addw r1, pc, #((1f - . - 4) | 1) /* 4 is the size of addw instruction itself */
+	ldr r1, =(1f + 1)
 	mov r2, sp
 	/* store everythig in jmpbuf */
-	stm r0, {r1-r2, r4-r11, lr}
+	stmia r0!, {r1-r2, r4-r11, lr}
+	/* save signal mask */
+	str r3, [r0]
 	mov r0, #0
 	bx lr
 1:
 	/* this is where we land after the jump
 	 * r0 is now pointing to signal mask
 	 * which we want to restore */
 	push {r1, r4, lr}
+	/* make sure SP is aligned to 8 bytes */
+	sub sp, #4
```

Comment:
You can do this in one operation quite easly - you push (and later pop) one register extra, e.g. r2:
`push {r1-r2, r4, lr}`